src/components/token/token.tsx

```diff
@@ -29,95 +40,137 @@ const shapeToClassMap: { [shape in TokenShape]: string } = {
 
 export const SHAPES = keysOf(shapeToClassMap);
 
+const fillToClassMap: { [fill in TokenFill]: string | null } = {
+  none: null,
+  light: 'euiToken--light',
+  dark: 'euiToken--dark',
+};
+
+export const FILLS = keysOf(fillToClassMap);
+
 const colorToClassMap: { [color in TokenColor]: string } = {
-  tokenTint01: 'euiToken--tokenTint01',
-  tokenTint02: 'euiToken--tokenTint02',
-  tokenTint03: 'euiToken--tokenTint03',
-  tokenTint04: 'euiToken--tokenTint04',
-  tokenTint05: 'euiToken--tokenTint05',
-  tokenTint06: 'euiToken--tokenTint06',
-  tokenTint07: 'euiToken--tokenTint07',
-  tokenTint08: 'euiToken--tokenTint08',
-  tokenTint09: 'euiToken--tokenTint09',
-  tokenTint10: 'euiToken--tokenTint10',
-  tokenTint11: 'euiToken--tokenTint11',
-  tokenTint12: 'euiToken--tokenTint12',
+  euiColorVis0: 'euiToken--euiColorVis0',
+  euiColorVis1: 'euiToken--euiColorVis1',
+  euiColorVis2: 'euiToken--euiColorVis2',
+  euiColorVis3: 'euiToken--euiColorVis3',
+  euiColorVis4: 'euiToken--euiColorVis4',
+  euiColorVis5: 'euiToken--euiColorVis5',
+  euiColorVis6: 'euiToken--euiColorVis6',
+  euiColorVis7: 'euiToken--euiColorVis7',
+  euiColorVis8: 'euiToken--euiColorVis8',
+  euiColorVis9: 'euiToken--euiColorVis9',
+  gray: 'euiToken--gray',
 };
 
 export const COLORS = keysOf(colorToClassMap);
 
-interface TokenProps {
+export interface TokenProps {
   /**
    * An EUI icon type
    */
   iconType: IconType;
   /**
-   * Size of the token
+   * For best results use one of the vis color names (or 'gray').
+   * Or supply your own color (can be used with dark or no fill only).
    */
-  size?: TokenSize;
+  color?: TokenColor | string;
+  /**
+   * Outer shape surrounding the icon
+   */
+  shape?: TokenShape;
+  /**
+   * `light` for lightened color with border, `dark` for solid, or `none`
+   */
+  fill?: TokenFill;
   /**
-   * By default EUI will auto color tokens. You can can however control it
-   * - `color`: can be `tokenTint01` thru `tokenTint10`
-   * - `shape`: square, circle, rectangle as options
-   * - `fill`: makes it a solid color
-   * - `hideBorder`: disables the outer border
+   * Size of the token
    */
-  displayOptions?: EuiTokenMapDisplayOptions;
+  size?: TokenSize;
```

Comment:
As `xs` isn't supported, this should probably be

```suggestion
  size?: Omit<TokenSize, 'xs'>;
```

and the logic below around `finalSize` could be removed.

I say "probably" as I'm not 100% sure we're not expecting a lot of passing through of props to this component which may complicate things.

Reply:
`TokenSize` establishes the overall size of the rendered token. But the logic for `finalSize` is for the Icon, not the Token.